14/homework_14.1.py

```diff
@@ -0,0 +1,22 @@
+
+# -----------------------------Домашнє завдання 14.1
+
+import re
+
+# регулярний вираз, який шукає послідовність, що шукає номер телефону.
+# він розділений на групи, які з'єднані між собою символом ?: на початку групи після (
+# (?:\B\+)? - шукає символ "+", він може бути відсутнім
+# (?:(?:\B\+)?38)? - в неї входить попередня група. Шукає код України +38
+# (?:\s?|-?) - пробіл та тире, може бути присутнім, або відсутнім. Повторюється 4 рази
+# (?:\(?0\d{2}\)?) - шукає три цифри, які починаються з нуля, та двух довільних
+# (?:\d{3}) - шукає три цифри
+# (?:\d{2}) - шукає дві цифри (повторюється 2 рази). В кінці стоїть \b що шукає границю "слова"
+tel_pat = r'(?:(?:\B\+)?38)?(?:\s?|-?)(?:\(?0\d{2}\)?)(?:\s?|-?)(?:\d{3})(?:\s?|-?)(?:\d{2})(?:\s?|-?)(?:\d{2})\b'
```

Comment:
1. Your pattern is overcomplicated. Try to make it simple.
2. Your pattern allows using of invalid patterns. For example:
```
l+380951234567 -> VALID
phone380953985657 -> VALID
+830951234567 -> VALID
+3809512345678 -> VALID
095123456 -> VALID
809512345678 -> VALID
3809512345678912 -> VALID
phone380951234567 -> VALID
a0951234567 -> VALID
8+30951234567 -> VALID
a+380951234567 -> VALID
+380951234567a -> VALID
+3809512345677a -> VALID
+3809512345677 -> VALID
80951234567 -> VALID
+830965254141 -> VALID
```

3. Implement this check for your phonebook, this is required by the task